src/regex.cr

```diff
@@ -257,6 +257,23 @@ class Regex
     #
     # Unsupported with PCRE.
     ENDANCHORED = 0x8000_0000
+
+    # Do not check the pattern for valid UTF encoding.
+    NO_UTF_CHECK = NO_UTF8_CHECK
+
+    # Enable matching against subjects containing invalid UTF bytes.
+    # Invalid bytes never match anything. The entire subject string is
+    # effectively split into segments of valid UTF.
+    #
+    # Read more in the [PCRE2 documentation](https://www.pcre.org/current/doc/html/pcre2unicode.html#matchinvalid).
+    #
+    # When this option is set, `MatchOptions::NO_UTF_CHECK` is ignored at match time.
```

Comment:
Perhaps it should also be noted the pattern must be valid UTF-8 in this case (which PCRE2 enforces)

Reply:
I don't think we need to mention that explicitly. The pattern should always be valid UTF. And this is also noted in the referenced PCRE2 documentation.